renovate.json

```diff
@@ -0,0 +1,68 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "config:recommended",
+    ":semanticCommits",
+    ":semanticCommitTypeAll(chore)"
+  ],
+  "labels": [
+    "dependencies"
+  ],
+  "reviewers": [
+    "team:gotify"
+  ],
+  "enabledManagers": [
+    "gomod",
+    "dockerfile",
+    "custom.regex"
+  ],
+  "customManagers": [
+    {
+      "customType": "regex",
+      "fileMatch": [
+        "^GO_VERSION$"
+      ],
+      "depTypeTemplate": "language",
+      "matchStrings": [
+        "^(?<currentValue>.+)\\n"
+      ],
+      "depNameTemplate": "https://github.com/golang/go",
+      "autoReplaceStringTemplate": "{{{newValue}}}",
+      "datasourceTemplate": "golang-version"
+    }
+  ],
+  "packageRules": [
+    {
+      "matchManagers": [
+        "gomod"
+      ],
+      "matchUpdateTypes": [
+        "minor",
+        "patch"
+      ],
+      "groupName": "Bump Go dependencies",
+      "groupSlug": "bump-dependencies-go"
+    },
+    {
+      "matchDepNames": [
+        "^https://github.com/golang/go$"
+      ],
+      "matchUpdateTypes": [
+        "minor",
+        "patch"
+      ],
+      "groupName": "Bump Go version",
+      "groupSlug": "bump-go-version"
+    },
+    {
+      "matchDepNames": [
+        "^https://github.com/golang/go$"
+      ],
+      "matchUpdateTypes": [
+        "major"
+      ],
+      "groupName": "Go version major bump",
+      "groupSlug": "bump-go-version-major"
```

Comment:
Any specific reason to split major and minor/patch go version updates? In my experience manual intervention is really rare for Go updates.

Reply:
No, just that the PR will look different, if it is not that useful in practice I can omit it.